tests/uri_test.c

```diff
@@ -0,0 +1,410 @@
+/*
```

Comment:
Do we/should we validate/test the character space (encoding) of the input buffer anywhere?

Reply:
We test that something url encoded still parses properly.

Reply:
But should we reject input that isn't encoded properly?